### PR TITLE
Implement PartialEq for Mat[2,3,4] and Vec[2,3,4]

### PR DESCRIPTION
**Pull request by MagicRB**
Implements PartialEq for Mat[2,3,4] and Vec[2,3,4]. I didn't do the wide types since I don't use them and it seems complicated

### CODE REVIEW

File: src/mat.rs [hunk 1]
```diff
@@ -1094,6 +1114,16 @@ macro_rules! mat4s {
 
 mat4s!(Mat4 => Rotor3, Bivec3, Vec4, Vec3, f32, Wat4 => WRotor3, WBivec3, Wec4, Wec3, f32x4);
 
+impl PartialEq for Mat4 {
+    fn eq(&self, other: &Self) -> bool {
+        self.cols[0] == other.cols[0] && self.cols[1] == other.cols[1] && self.cols[2] == other.cols[2] && self.cols[3] == other.cols[3]
+    }
+
+    fn ne(&self, other: &Self) -> bool {
+        self.cols[0] != other.cols[0] || self.cols[1] != other.cols[1] || self.cols[2] != other.cols[2] || self.cols[3] != other.cols[3]
```
Comment (termhn) [inline src/mat.rs]: could you format these (and the rest) like

```rust
expression
|| nextexpr
|| nextexpr
```
etc

Comment (MagicRB) [reply src/mat.rs]: now it's all formatted by rustfmt, oh and sorry for not doing it sooner, I am a bit sick

**Comment by MagicRB** [comment]
Sorry for not handling this sooner but I was sick, anyway, I ran rustfmt on it, so it should all be compliant.

**Comment by termhn** [comment]
Closed in favor of #38 as well, thanks again @MagicRB :)

**Pull request closed without merging**